controller/githubController.js

```diff
@@ -0,0 +1,112 @@
+const http = require('http-status-codes')
+const statusMsg = require('../config/statusMsg')
+const tokenGenerator = require('./authTokenGenerator')
+dotenv.config({
+    path: './config/.env'
+})
+
+exports.github = async (req, res, next) => {
+    try {
+        github_access_token = req.body.access_token
+        await request(`https://api.github.com/user?access_token=${github_access_token}`, {
+            headers: {
+                'user-agent': 'Mentors'
+            }
+        },
+            async (err, response) => {
+                const userInfo = JSON.parse(response.body)
+                const name = userInfo.name.split(' ', 2)
+                const data = {
+                    id: userInfo.id,
+                    first_name: name[0],
+                    last_name: name[1],
+                    email: userInfo.email,
+                }
+                const access_token = await tokenGenerator.access_token(userInfo.email)
+                const refresh_token = await tokenGenerator.refresh_token(userInfo.email)
+                const payload = {
+                    "accessToken": access_token,
+                    "refreshToken": refresh_token,
+                    "data": data
+                }
+                try {
+                    if (await dbQuery.idExists(data, refresh_token)) {
+                        return res.status(http.OK).json({
+                            "success": statusMsg.success.msg,
+                            "payload": payload
+                        })
+                    }
+                    else if (await dbQuery.emailExists(data)) {
+                        await dbQuery.updateWithId(data, refresh_token)
+                        return res.status(http.OK).json({
+                            "success": statusMsg.success.msg,
+                            "payload": payload
+                        })
+                    }
+                    else if (!userInfo.email) {
+                        return res.status(http.CONFLICT).json({
+                            "success": statusMsg.fail.msg,
+                            "payload": data,
+                            "error": {
+                                "code": http.CONFLICT,
+                                "message": http.CONFLICT
+                            }
+
+                        })
+                    }
+                    else {
+                        await dbQuery.createAccount(data, refresh_token)
+                        return res.status(http.CREATED).json({
+                            "success": statusMsg.success.msg,
+                            "payload": payload
+                        })
+                    }
+                }
+                catch (err) {
+                    err.status = http.INTERNAL_SERVER_ERROR
+                    next(err)
+                }
+            }
+        )
+    }
+    catch (err) {
+        err.status = http.CONFLICT
+        next(err)
+    }
+}
+
+
+const dbQuery = {
```

Comment:
@AwaleRohin @vipul- guys create the reusable code for this purpose. You guys are repeating the same content on every authentication part.

Reply:
fixed